ecosystem/nft-metadata-crawler/src/asset_uploader/worker/mod.rs

```diff
@@ -0,0 +1,128 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    asset_uploader::worker::config::AssetUploaderWorkerConfig, config::Server,
+    utils::constants::MAX_ASSET_UPLOAD_RETRY_SECONDS,
+};
+use anyhow::Context;
+use axum::{
+    body::Body, http::StatusCode, response::IntoResponse, routing::post, Extension, Json, Router,
+};
+use reqwest::{multipart::Form, Client};
+use serde::Deserialize;
+use std::{sync::Arc, time::Duration};
+use tracing::{error, info};
+use url::Url;
+
+pub mod config;
+
+#[derive(Clone)]
+pub struct AssetUploaderWorkerContext {
+    config: Arc<AssetUploaderWorkerConfig>,
+}
+
+#[derive(Debug, Deserialize)]
+struct UploadRequest {
+    url: Url,
+}
+
+impl AssetUploaderWorkerContext {
+    pub fn new(config: AssetUploaderWorkerConfig) -> Self {
+        Self {
+            config: Arc::new(config),
+        }
+    }
+
+    /// Uploads an asset to Cloudflare and returns the response
+    async fn upload_asset(&self, url: &Url) -> anyhow::Result<impl IntoResponse> {
+        let hashed_url = sha256::digest(url.to_string());
+        let client = Client::builder()
+            .timeout(Duration::from_secs(MAX_ASSET_UPLOAD_RETRY_SECONDS))
```

Comment:
should this live in the config instead of const var? we might not know right away what's the best seconds to retry for

Reply:
This is just an arbitrarily high number. Theoretically, we should be nowhere near this limit because of the size limit for cloudflare.